controllers/users.js

```diff
@@ -1,33 +1,41 @@
+import dotenv from "dotenv";
+import { hashSync, genSaltSync } from "bcrypt";
 import models from "../models";
 
+dotenv.config();
 const { User } = models;
 
 class UserController {
-  /**
-   * Fetch a single user's information.
-   * @param  {Object} req - The request object.
-   * @param  {Object} res - The response object.
-   * @return {Object} The request response
-   */
-  static findOne(req, res) {
-    const { userId } = req.params;
-    User.findOne({ where: { id: userId } })
-      .then(result => {
-        if (result) {
-          return res.status(200).json({
-            message: "User retrieved successfully",
-            user: result.dataValues
-          });
+  static async signUp(req, res) {
+    const { email, password, username } = req.body;
+    try {
```

Comment:
Why try and catch and not the promise style??
what is the best way??

Reply:
We are supposed to use async and await